extensions/import/src/controllers/mainController.ts

```diff
@@ -25,15 +25,23 @@ export default class MainController extends ControllerBase {
 	public deactivate(): void {
 	}
 
-	public activate(): Promise<boolean> {
-		const outputChannel = vscode.window.createOutputChannel(constants.serviceName);
-		new ServiceClient(outputChannel).startService(this._context);
-
-		managerInstance.onRegisteredApi<FlatFileProvider>(ApiType.FlatFileProvider)(provider => {
-			this.initializeFlatFileProvider(provider);
+	public async activate(): Promise<boolean> {
+		return new Promise<boolean>(async (resolve) => {
+			try {
+				/*
+				Registering the command before downloading the file to avoid the command not found error.
+				*/
+				managerInstance.onRegisteredApi<FlatFileProvider>(ApiType.FlatFileProvider)(provider => {
+					this.initializeFlatFileProvider(provider);
+					resolve(true);
+				});
+				const outputChannel = vscode.window.createOutputChannel(constants.serviceName);
+				await new ServiceClient(outputChannel).startService(this._context);
+			} catch (error) {
```

Comment:
You don't need to do this, it'll just reject() on its own.

Reply:
And doing it like this will cause it to ignore the error - we want this to fail the activation if an exception is thrown.